hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@horsburgh @dtarb @selimnairb I have implemented what @horsburgh suggested. In short, the resource will be published even if there is a failure with metadata deposition from CrossRef, so users will not be aware of the failure unless hydroshare support contacts them and they always see their resources are published regardless after they clicked on "Publish" button. The crontab celery task runs once every midnight to collect pending published resources and notify hydroshare support by email any pending published resources with detailed response content from CrossRef written to the log for perusal if needed. From the admin interface, the admin can see the pending published resource page including DOI field and modify metadata as needed in the same way as for any HydroShare resources. I think it is better to send an aggregated email to hydroshare support one time each day to notify support of current status with any pending published resources including delinquent published resources as well as those pending DOI activation. I don't know how to add more info to admin console/page regarding most recent DOI acquisition attempt, in fact, after some investigation with Django and Mezzanine out-of-box admin customization, I am not even sure this can be done with the out-of-box admin. @selimnairb Other than this, I believe I have addressed all your code review comments. @selimnairb Can you give a quick code review go-through to see if you can give a +1 now.